README.md

```diff
@@ -3,18 +3,281 @@
+	klog.Infof("respWithStruct: %s", respWithStruct.Resp)
+}
+```
+
+Important notes:
+1. Each dubbo Interface corresponds to a DubboCodec. Please do not configure multiple clients sharing a single DubboCodec.
+
+#### initializing server
 
-- [hessian serialization](http://hessian.caucho.com/doc/hessian-serialization.html)
+```go
+import (
+	"github.com/cloudwego/kitex/server"
+	dubbo "github.com/kitex-contrib/codec-dubbo/pkg"
+	hello "github.com/kitex-contrib/codec-dubbo/samples/helloworld/kitex/kitex_gen/hello/greetservice"
+	"log"
+	"net"
+)
+
+func main() {
+	// server address to listen on
+	addr, _ := net.ResolveTCPAddr("tcp", ":21000")
+	svr := hello.NewServer(new(GreetServiceImpl),
+		server.WithServiceAddr(addr),
+		// configure DubboCodec
+		server.WithCodec(dubbo.NewDubboCodec(
+			// Interface Name of kitex service. Other dubbo clients and kitex clients can refer to this name for invocation.
+			dubbo.WithJavaClassName("org.cloudwego.kitex.samples.api.GreetProvider"),
+		)),
+	)
+
+	err := svr.Run()
+
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+```
+
+Important notes:
+1. Each Interface Name corresponds to a DubboCodec. Please do not configure multiple servers sharing a single DubboCodec.
+
+## Benchmark
+
+### Benchmark Environment 
+CPU: **Intel(R) Xeon(R) Gold 5118 CPU @ 2.30GHz**  
+Memory: **192GB**
+
+### Benchmark Code
+
+Referring to [dubbo-go-benchmark](https://github.com/dubbogo/dubbo-go-benchmark). Converting dubbo client and dubbo server
+to kitex client and kitex server. Please see [this](https://github.com/kitex-contrib/codec-dubbo/tree/main/tests/benchmark).
+
+### Benchmark Result
+
+#### dubbo -> dubbo
+
+```shell
+bash deploy.sh dubbo_server -p 21003
+bash deploy.sh dubbo_client -p 21004 -addr "127.0.0.1:21003"
+bash deploy.sh stress -addr '127.0.0.1:21004' -t 1000000 -p 100 -l 256
+```
+
+Result:  
+
+| average rt |  tps  | success rate |
+|:----------:|:-----:|:------------:|
+|  2659350   | 36974 |   1.000000   |
+|  2655246   | 38500 |   1.000000   |
+|  2511988   | 38489 |   1.000000   |
+|  2615031   | 37960 |   1.000000   |
+
+Resource:
+
+|   process_name    | %CPU  | %MEM |
+|:-----------------:|:-----:|:----:|
+| dubbo_client_main | 1807  | 0.1  |
+| dubbo_server_main | 792.1 | 0.0  |
+|    stress_main    | 904.6 | 0.1  |
+
+#### kitex -> kitex
+
+```shell
+bash deploy.sh kitex_server -p 21001
+bash deploy.sh kitex_client -p 21002 -addr "127.0.0.1:21001"
+bash deploy.sh stress -addr '127.0.0.1:21002' -t 1000000 -p 100 -l 256
+```
+
+Result:
+
+| average rt |  tps  | success rate |
+|:----------:|:-----:|:------------:|
+|  2310628   | 46015 |   1.000000   |
+|  2363729   | 44202 |   1.000000   |
+|  2256177   | 43280 |   1.000000   |
+|  2194147   | 43171 |   1.000000   |
+
+Resource:
+
+|   process_name    | %CPU  | %MEM |
+|:-----------------:|:-----:|:----:|
+| kitex_client_main | 914.6 | 0.0  |
+| kitex_server_main | 520.5 | 0.0  |
+|    stress_main    | 1029  | 0.1  |
+
+### Benchmark Summary
```

Comment:
So we'd better also remove this comparison

Reply:
Maybe we can write here:

> Since the dubbo-go-hessian2 relies on reflect to to encoding/decoding, there's great potential to improve the performance with a codec based on generated Go code. A fastCodec for Hessian2 (#46) is planned for better performance.